core/templates/dev/head/services/ContextService.ts

```diff
@@ -96,6 +105,35 @@ oppia.factory('ContextService', [
         return (this.getPageContext() === PAGE_CONTEXT.QUESTION_EDITOR);
       },
 
+      getEntityId: function() {
+        var pathnameArray = UrlService.getPathname().split('/');
+        for (var i = 0; i < pathnameArray.length; i++) {
+          if (pathnameArray[i] === 'embed' ||
+              pathnameArray[i] === 'story_editor') {
+            return pathnameArray[i + 2];
+          }
+        }
+        return pathnameArray[2];
+      },
+
+      getEntityType: function() {
+        var pathnameArray = UrlService.getPathname().split('/');
+        for (var i = 0; i < pathnameArray.length; i++) {
+          if (pathnameArray[i] === 'create' || pathnameArray[i] === 'explore') {
+            return 'exploration';
```

Comment:
Make as many of these strings as possible into named constants. Use these named constants in callers of getEntityType().

Reply:
Done